app/src/main/java/com/github/alfabravo2013/readyforexams/domain/login/LocalDataSource.kt

```diff
@@ -0,0 +1,43 @@
+package com.github.alfabravo2013.readyforexams.domain.login
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import com.github.alfabravo2013.readyforexams.util.Result
+
+class LocalDataSource {
+    private val registeredUsers = mutableMapOf<String, String>()
+
+    init {
+        registeredUsers["test@test.com"] = "123456789"
+    }
+
+    suspend fun signUp(email: String, password: String): Result =
+        withContext(Dispatchers.IO) {
```

Comment:
no dispatchers here. I would suggest to keep things simple for now. If you want to do things in background thread right now do it inside the ViewModel.